examples/evm/src/main.rs

```diff
@@ -86,7 +90,7 @@ async fn main() {
         .add_input(&to_vec(&zkdb).unwrap())
         .build();
     let mut exec = Executor::from_elf(env, EVM_ELF).unwrap();
-    let session = exec.run().unwrap();
+    let session = exec.run_with_callback(|segment| Ok(Box::new(FileSegmentRef::new(&segment, &temp_dir())?))).unwrap();
```

Comment:
I think this calls temp_dir() each callback so each segment would get its own new directory. I would consider calling it once before run_with_callback and then just passing a reference on each callback.